x-pack/solutions/observability/plugins/synthetics/server/cases/suggestion.ts

```diff
@@ -0,0 +1,253 @@
+              },
+            },
+            aggs: {
+              by_monitor: {
+                terms: {
+                  field: 'config_id',
+                  // TODO: TBD
+                  // it limits suggestions to 5 monitors
+                  size: 5,
+                },
+                aggs: {
+                  latest_run: {
+                    top_hits: {
+                      sort: [
+                        {
+                          '@timestamp': {
+                            order: 'desc',
+                          },
+                        },
+                      ],
+                      // TODO: Limit the suggestion to only the latest run per monitor
+                      // it gives random location at each run
+                      size: 1,
+                      _source: {
+                        includes: [
+                          'monitor.name',
+                          'monitor.type',
+                          'monitor.status',
+                          'observer.geo.name',
+                          'observer.name',
+                          'service.name',
+                          'monitor.id',
+                          'config_id',
+                          'url.full',
+                          '@timestamp',
+                          'meta.space_id',
+                        ],
+                      },
+                    },
+                  },
+                },
+              },
+            },
+          });
+
+          const monitors =
+            (results.aggregations as SyntheticsAggregationsResponse['aggregations'])?.by_monitor
+              .buckets || [];
+          if (monitors.length === 0) {
+            logger.debug(`No Synthetics monitors found for service ${serviceNames.join(', ')}`);
+            return { suggestions: [] };
+          }
+          const savedObjectsAttrHash: Record<string, EncryptedSyntheticsMonitorAttributes> = {};
+          const encryptedSavedObjectsClient = encryptedSavedObjects.getClient();
+          const monitorConfigRepository = new MonitorConfigRepository(
+            scopedSavedObjectsClient,
+            encryptedSavedObjectsClient
+          );
+          const configIdsFilter = getSavedObjectKqlFilter({
+            field: 'config_id',
+            values: monitors.map((mon) => mon.latest_run.hits.hits[0]._source.config_id),
+          });
+
+          const resultMonitorsConfigWithLegacy =
+            await monitorConfigRepository.find<EncryptedSyntheticsMonitorAttributes>({
+              filter: configIdsFilter,
+              perPage: 10,
+            });
+
+          if (resultMonitorsConfigWithLegacy.total === 0) {
+            logger.error(`No Synthetics SavedObjects found for the related tests runs`);
+            return { suggestions: [] };
+          }
+
+          for (const savedObj of resultMonitorsConfigWithLegacy.saved_objects) {
+            if (!savedObj.error && savedObj.id && savedObj.attributes) {
```

Comment:
nit:
```suggestion
            if (!savedObj.error) {
```

`id` and `attributes` are always defined